rust/kcl-lib/src/unparser.rs

```diff
@@ -30,108 +30,125 @@ impl Program {
     }
 
     pub fn recast(&self, buf: &mut String, options: &FormatOptions, indentation_level: usize) {
-        let indentation = options.get_indentation(indentation_level);
-
         if let Some(sh) = self.shebang.as_ref() {
             write!(buf, "{}\n\n", sh.inner.content).no_fail();
         }
 
-        if self
-            .non_code_meta
-            .start_nodes
-            .iter()
-            .any(|noncode| !matches!(noncode.value, NonCodeValue::NewLine))
-        {
-            for start in &self.non_code_meta.start_nodes {
-                let noncode_recast = start.recast(options, indentation_level);
-                buf.push_str(&noncode_recast);
-            }
+        recast_body(
+            &self.body,
+            &self.non_code_meta,
+            &self.inner_attrs,
+            buf,
+            options,
+            indentation_level,
+        );
+    }
+}
+
+fn recast_body(
+    items: &[BodyItem],
+    non_code_meta: &NonCodeMeta,
+    inner_attrs: &NodeList<Annotation>,
+    buf: &mut String,
+    options: &FormatOptions,
+    indentation_level: usize,
+) {
```

Comment:
The code in this function didn't change semantically. I just factored it out to a function.